.evergreen/serverless/create-instance.sh

```diff
@@ -46,6 +51,12 @@ if [ -z "$SERVERLESS_INSTANCE_NAME" ]; then
     SERVERLESS_INSTANCE_NAME="$RANDOM-DRIVERTEST"
 fi
 
+DIR=$(dirname $0)
```

Comment:
This line was just moved up from later in the file, as the current directory was previously used for invoking `get-instance.sh`.

I noted in the commit message that we could have used `${BASH_SOURCE[0]}` here but I don't think it makes any difference since this file is never sourced (related reading: https://stackoverflow.com/questions/35006457/choosing-between-0-and-bash-source).